### DISCUSSION TITLE
How to use just JavaScript with next.js without JSX

### DISCUSSION THREAD

**Discussion by Kreijstal** [vercel/next.js#44000] [Help]
I want to use standard node.js JavaScript without JSX, react documentation says you can simply use createElement, but that's for react, how does that work for next.js? I don't see it anywhere I googled it.

**Comment by icyJoseph** [comment]
JSX is not necessarily needed for React, nor Next.js, that's correct. 

I am not sure why you'd want this, and it is quite worrying in a few ways.

Do remember though, that since React 17, the pragma is no longer, always `createElement`, rather it might be an optimised version, https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html, which you'd be missing on.

You can even make your own pragma! Check this other answer I gave https://github.com/vercel/next.js/discussions/37802#discussioncomment-2976333

Anyway, all you need to do is, for example:

```jsx
import { createElement, Fragment } from "react";
import { type NextPage } from "next";
import Head from "next/head";

const About: NextPage = () => {
  return createElement(
    Fragment,
    null,
    createElement(
      Head,
      null,
      createElement("title", null, "About"),
      createElement("meta", { name: "name", content: "meta tag content" })
    ),
    createElement("div", null, createElement("h1", null, "Hello world"))
  );
};

export default About;
```

And there you go. Scary though. Not judging! I suspect there might some use case that you found doesn't work? Next.js is a React meta framework, so React techniques just work on Next.js

**Reply by Kreijstal** [reply]
I needed this to run on browser code, there's always JavaScript tagged templates so you can write pseudo-jsx, but without compilation step. (Which has benefits)

**Reply by icyJoseph** [reply]
> I needed this to run on browser code, 

What do you mean exactly? Next.js transpiles all of the code to calls to _jsx or createElement already.
You can also load babel and make it transpile a script tag with type `text/babel`, like [this](https://github.com/icyJoseph/vanilla-react-redux/blob/master/vanilla-react/index.html).

**Reply by Kreijstal** [reply]
Yes, that is exactly the problem, there's transpilation overhead, so I desire no transpilation step. Nor compilation.
